File: Practica 4/CivitasJava4/CivitasJava/src/civitasjava/SorpresaPorCasaHotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitasjava;

import java.util.ArrayList;

/**
 *
 * @author francisco
 */
public class SorpresaPorCasaHotel extends Sorpresa {
    
    private int valor; // Guarda cuanto se paga por cada casa u hotel
    
    SorpresaPorCasaHotel(int valor, String texto){
        super(texto);
        
        this.valor = valor;
    }
    
    
    /**
     * @brief Modifica el saldo del jugador multiplicando valor con la
     *        cantidad de casas y hoteles que se poseen
     * @param actual, el indice del jugador
     * @param todos, el array de jugadores
     * @return void
     */
    @Override
    void aplicarAJugador (int actual, ArrayList<Jugador> todos){ 
        
      if( jugadorCorrecto(actual, todos) ){
        this.informe(actual, todos);
        
        Jugador jugador = todos.get(actual);
        jugador.modificarSaldo(this.valor * jugador.cantidadCasasHoteles());
      }
    }
}

File: Practica 4/CivitasRuby4/CivitasRuby/pkg/RubyApplication1-0.0.1/lib/sorpresa_por_casa_hotel.rb
module Civitas

class Sorpresa_por_casa_hotel < Sorpresa

  def initialize(valor, texto)
    super(texto)
    
    @valor = valor
  end
  
# @brief Aplica la sorpresa de modificar el saldo multiplicar @valor con la
#   cantidad de casas y hoteles que el jugador posee
# @param actual, el indice del jugador
# @param todos, el vector de jugadores
# @return Es void
def aplicar_a_jugador_por_casa_hotel(actual,todos)
  if(jugador_correcto(actual,todos))
    informe(actual,todos)
    todos[actual].modificar_saldo(@valor * todos[actual].cantidad_casas_hoteles())
  end
end

end
end
